lib/dust.js

```diff
@@ -565,9 +585,36 @@
     return this;
   };
 
+  /**
+   * Pipes to a WritableStream. Note that backpressure isn't implemented,
+   * so we just write as fast as we can.
+   * @param stream {WritableStream}
+   * @return self
+   */
   Stream.prototype.pipe = function(stream) {
+    if(typeof stream.write !== 'function' ||
+       typeof stream.end !== 'function') {
+      dust.log('Incompatible stream passed to `pipe`', WARN);
+      return this;
+    }
+
+    var destEnded = false;
+
+    if(typeof stream.emit === 'function') {
```

Comment:
makes sense. asking for consistency rather then for more features. i agree though keeping it lax make sense. 

The only thing (very nit picky) is then the choice of the name `chunk.prototype.stream`.  Would calling it stream be confusing? `await` makes sense in promises-land... maybe `awaitStream`? idk. very nit picky so i'm cool with what's currently here.